### ISSUE TITLE
Unable to interact with any element not currently visible since Dynamics Update

### ISSUE THREAD

**Issue by ryanpiran** [Microsoft/EasyRepro#489]
Hi, 

I'm running Dynamics 365 (9.1.0.5024) online with EasyRepro V9.1.0.1. We've recently begun experiencing an issue where we get errors when interacting with any elements which are not currently visible. We've discovered that a recent update of dynamics has introduced 'lazy loading' and the objects not currently visible therefore do not exist. Previously we were able to use a command such as :

`Xrm.App.Entity.SetValue("title", "Example String");`

Where 'title' is not currently visible and EasyRepro would navigate to this element and then continue the script.

**Comment by jlattimer** [comment]
Wouldn't you want this to be the case? After all, if a user can't see an element they shouldn't be able to interact with it.

**Comment by ryanpiran** [comment]
@jlattimer The scenario would be when an element is on the current page but not in view as the user would need to scroll down the page to get to it. Previously a single `Xrm.App.SetValue` call to the element out of view (but on the current page) would cause the script to navigate to the element before setting it. However this no longer functions.

**Comment by BenjaminDeveloper** [comment]
Has there been any progress on this?

**Comment by jlattimer** [comment]
My existing orgs. are only 9.1.0.4834 and didn't exhibit this behavior. I spun up a trial and didn't and still ended up with 4834 so I'll try and circle back to this later if nobody has gotten to it by then.

**Comment by jlattimer** [comment]
My orgs just got updated and I can reproduce this - working on a fix.

**Comment by Bala-Ramadoss** [comment]
@jlattimer - Please let know if there is any update on the fix. Waiting for the fix to be available. thank you

**Comment by TYLEROL** [comment]
@BenjaminDeveloper / @ryanpiran / @Bala-Ramadoss - I've been collaborating behind the scenes with Jason on this one. I have a meeting scheduled with our internal development team that introduced this change in behavior for the Unified Interface.

After my discussion with them we'll determine the best path for fixing this in the long term. Thanks for your patience while we work to resolve this new rendering pattern with the Unified Interface.

**Comment by jlattimer** [comment]
In the meantime - you could add this to you test after the page loads - it will check to see if the page has a scrollbar, if so it will scroll to the bottom and them back to the top forcing a complete load.

```csharp
client.Browser.Driver.ExecuteScript(@"if (document.getElementById('tab-section').scrollHeight > document.getElementById('tab-section').clientHeight) {
    document.getElementById('tab-section').scrollTop = document.body.scrollHeight;
    setTimeout(function() {document.getElementById('tab-section').scrollTop = 0;}, 150);}");
```

**Comment by TYLEROL** [comment]
@BenjaminDeveloper / @ryanpiran  / @Bala-Ramadoss / @jlattimer - We've merged in an update to the develop branch in an effort to address this behavior. This change introduces a new browser option called **UCITestMode**. When this option is set to True, we will enable a flag that puts the UCI app into test mode. This test mode disables the lazy loading functionality that was introduced in a recent CDS update. The default behavior will be to set this value to True.

This should allow the EasyRepro APIs to function as originally intended by forcing UCI to load all DOM elements on the page, regardless of page location.

I am currently running some tests internally to make sure we don't encounter any other side effects of this change, but if my tests pass we plan to release an update by EOD tomorrow. **Please feel free to pull down the latest changes from the _develop_ branch and let us know if you encounter any problems with this new functionality**

**Note: This option is not intended for performance testing as it will increase load times slightly (due to larger rendering patterns) in comparison to what users will see in a normal day-to-day environment. We recommend setting _UCITestMode_ to False if using EasyRepro to generate performance based runs**

Warm Regards,
Tyler

**Comment by TYLEROL** [comment]
The change to fix this was made available last week in the latest releases/v9.1 branch and via the 9.1.0.5230 NuGet package. Please let us know via new issues if you have any problems with the latest release.

Thanks,
Tyler
